pmlc/target/x86/heatmap.tpl.cc

```diff
@@ -22,4 +30,67 @@ float kHeatmapValues[] = {
     // clang-format on
 };
 
+static std::map<std::tuple<unsigned, unsigned, unsigned>, double> *
```

Comment:
So instead of adding new helper functions to the c++ template file, I think we should add it to a normal c++ file.